crates/rustc_codegen_c_ast/src/ty.rs

```diff
@@ -0,0 +1,177 @@
+//! This module defines the AST nodes for C types.
+
+use rustc_data_structures::intern::Interned;
+use rustc_type_ir::{IntTy, UintTy};
+
+use crate::expr::CValue;
+use crate::pretty::{Print, PrinterCtx};
+use crate::ModuleCtx;
+
+/// C types.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub enum CTy<'mx> {
+    /// A primitive C type.
+    Primitive(CPTy),
+    /// A non-primitive C type, e.g. a pointer type.
+    Ref(Interned<'mx, CTyKind<'mx>>),
+}
+
+/// C primitive types.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub enum CPTy {
```

Comment:
Would a name like `CIntTy` be more fitting here, or will floats eventually be added? I think it probably makes sense longer term to keep integers and floats separate since they can wind up being used pretty differently, like Rust's IR structures do https://github.com/rust-lang/rust/blob/f7b5e5471b79a556dfd758a25265a777fbdad7ac/compiler/rustc_type_ir/src/ty_kind.rs#L70

Reply:
There will be `void`, `bool` and `char`: https://github.com/Wybxc/rustc_codegen_c_ospp/blob/ebb3fc786261dc0b7927f05d0d34b725bfffb449/crates/rustc_codegen_c_ast/src/type.rs#L111

Reply:
It seems like there is less benefit to keep primitives distinct from other types, compared to grouping similar types like integers. Could these all move to `CTy`? Something like:

```rust
pub enum CTy {
    Void,
    Bool,
    Char,
    Int(CIntTy),
    UInt(CUintTy),
    Pointer(CTy<'mx>),
    Ref(Interned<'mx, CTyKind<'mx>>),
}

pub enum CIntTy {
    I32,
    // ...
}

// Could also just go under `CIntTy`
pub enum CUnitTy {
    U32,
    // ...
}
```

This would be a bit more consistent with Rust's `TyKind`.

Reply:
> This would be a bit more consistent with Rust's `TyKind`.

One other note here is the docs for `CTy` and `CTyKind` should make the differences clear since they seem likely to overlap (interning?)